File: CSharpAnalyzerMcp/Models/Roslyn/DiagnosticInfo.cs
namespace CSharpAnalyzerMcp.Models.Roslyn;

public class DiagnosticInfo
{
    public string Id { get; set; } = string.Empty;
    public string Severity { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public int Line { get; set; }
    public int Column { get; set; }
    public int EndLine { get; set; }
    public int EndColumn { get; set; }
    public string FilePath { get; set; } = string.Empty;
}

File: python-analyzer-mcp/src/models/diagnostic_info.py
"""Data models for Python analyzer"""
from dataclasses import dataclass
from typing import Optional


@dataclass
class DiagnosticInfo:
    """Information about a diagnostic (error, warning, etc.)"""
    message: str
    category: str
    code: Optional[str] = None
    file: Optional[str] = None
    line: Optional[int] = None
    column: Optional[int] = None
    severity: str = "error"
